client/api/errors.proto

```diff
@@ -13,3 +13,21 @@ message CmdletError {
     // Human-readable error message - can be empty.
     string message = 3;
 }
+
+// WindowsApiError details errors yielded by direct system calls to the Windows API.
+message WindowsApiError {
+    // Name of the Windows API procedure that returned an error.
+    string procedure_name = 1;
+
+    // Error code that got returned. For additional details of a code please refer 
+    // to https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes
+    uint32 code = 2;
+
+    // Windows API's documentation lists hexadecimal error codes; this is the hexadecimal
+    // representation of the error code. For additional details, please refer
+    // to https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/705fb797-2175-4a90-b5a3-3918024b10b8
```

Comment:
Might be worth linking to https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d too?